src/transformers/models/cpmant/modeling_cpmant.py

```diff
@@ -0,0 +1,1090 @@
+        key = key.view(batch_size, len_k, self.num_heads, self.dim_head).permute(0, 2, 1, 3)
+        value = value.view(batch_size, len_k, self.num_heads, self.dim_head).permute(0, 2, 1, 3)
+
+        if past_kv is not None:
+            key = torch.cat([past_kv[0], key], dim=-2)
+            value = torch.cat([past_kv[1], value], dim=-2)
+            len_k = key.size(-2)
+
+        # (b, n_h, len_q, d_h) @ (b, n_h, d_h, len_k) -> (b, n_h, len_q, len_k)
+        score = torch.matmul(query, key.transpose(-1, -2)) / math.sqrt(self.dim_head)
+        score = score + position_bias
+
+        score = torch.masked_fill(
+            score,
+            attention_mask.view(batch_size, 1, len_q, len_k) == torch.tensor(False),
+            torch.scalar_tensor(float("-inf"), device=score.device, dtype=score.dtype),
+        )
+        score = self.softmax(score)
+
+        score = torch.masked_fill(
+            score,
+            attention_mask.view(batch_size, 1, len_q, len_k) == torch.tensor(False),
+            torch.scalar_tensor(0, device=score.device, dtype=score.dtype),
+        )
+
+        if self.dropout is not None:
+            score = self.dropout(score)
+
+        # (b, n_h, len_q, len_k) @ (b, n_h, len_k, d_h) -> (b, n_h, len_q, d_h)
+        score = torch.matmul(score, value)
+
+        score = score.view(batch_size, self.num_heads, len_q, self.dim_head).permute(0, 2, 1, 3)
+        score = score.contiguous().view(batch_size, len_q, self.num_heads * self.dim_head)
+
+        score = self.attention_out(score)
+
+        if self.use_cache:
+            return score, (key, value)
+
+        return score
+
+
+class CPMAntSelfAttentionBlock(nn.Module):
+    """The whole cross-attention block. A sequence of operation. Consists of layernorm, self-attention and residual
+    connection.
+
+        Args:
+            dim_model (int): Main dimension of modules in transformer blocks.
+            num_heads (int): Number of attention heads in the Transformer encoder.
+            dim_head (int): Dimension of attention heads for each attention layer in the Transformer encoder.
+            eps (float, optional): The epsilon used by the layer normalization layers.
+            dropout_p (float, optional): Defaults to 0.
+            use_cache (bool, optional): Whether to use cache.
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        dim_model: int,
+        num_heads: int,
+        dim_head: int,
+        eps: float = 1e-6,
+        dropout_p: Optional[float] = None,
+        use_cache: Optional[bool] = False,
+    ):
+        super().__init__()
+        self.use_cache = use_cache
+        self.layernorm_before_attention = CPMAntLayerNorm(
+            dim_norm=dim_model,
+            eps=eps,
+        )
+        self.self_attention = CPMAntAttention(
+            dim_model=dim_model,
+            num_heads=num_heads,
+            dim_head=dim_head,
+            dropout_p=dropout_p,
+            use_cache=self.use_cache,
+        )
```

Comment:
you can pass `use_cache` directly to the `forward` methods, that is usually how we do it in `transformers`